### ISSUE TITLE
Latest knative.dev/pkg adds support for conversion webhooks

### ISSUE THREAD

**Issue by mattmoor** [tektoncd/pipeline#2047]
This is at HEAD and will be in 0.13.

cc @vdemeester

**Comment by bobcatfish** [comment]
Status: we have bumped, but we haven't yet updated our webhooks yet.

**Comment by riceluxs1t** [comment]
This may not be the best place to ask but @vdemeester could you point me to a link that describes what conversion web hook does? I have some code that creates a simple PipelineRun object that explicitly specifies its apiVersion to be `v1beta1` but when I use a Dynamic client (or Tekton client) to ask "give me all the PipelineRuns of apiVersion `v1alpha1`, the client returns the pipeline run I created (I expect no pipeline run to be returned). The opposite also seems to happen ; that is, I create a PR with apiVersion equal to `v1alpha` and ask "give me all PRs with apiVrsion = `v1beta`, I get back the PR (and I don't expect to)

**Comment by mattmoor** [comment]
When an API serves multiple versions, e.g. v1alpha1 and v1beta1, resources exist at all versions.

Without conversion webhooks, what you'd see is the exact same resource being returned with different TypeMeta (e.g. you'd get the v1alpha1 or v1beta1 that you asked for).

Conversion webhooks let us change the schema of resources across versions, so if fields move between alpha and beta we can return them in the appropriate place based on the version at which folks are asking for them.

So don't think "I created a v1beta1 PipelineRun", but "I created a PipelineRun, using a v1beta1 schema".

Hopefully that makes sense?
